include/swift/Remote/MetadataReader.h

```diff
@@ -389,6 +389,90 @@ class MetadataReader {
     ContextDescriptorCache.clear();
   }
 
+  void dumpAsFullMetadata(RemoteAddress MetadataAddress) {
+    auto offset = uint64_t(sizeof(StoredSize));
+
```

Comment:
Side note: I think this is too much code to put into a header file. I would move this into a .cpp file.
Also, if this linking against LLVM (not sure), you may want to use http://llvm.org/doxygen/classllvm_1_1raw__ostream.html for the formatting.